lab-shawn/server.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const net = require('net');
+const EE = require('events');
+const Client = require('./model/client.js');
+const PORT = process.env.PORT || 3000;
+const server = net.createServer();
+const ee = new EE();
+
+const pool = [];
+
+server.listen(PORT, function(){
+  console.log(`Served Up: ${PORT}`);
+});
+
+server.on('connection', function(socket){
+  var client = new Client(socket);
+  pool.push(client);
+  console.log('successfully connected');
+
+  socket.on('data', function(data){
+    const command = data.toString().split(' ').shift().trim();
+
+    if(command.startsWith('@')){
+      ee.emit(command,client,data.toString().split(' ').slice().join(' '));
+      return;
+    }
+
+    ee.emit('default',client, data.toString());
+  });
+
+  socket.on('error', function(err){
+    if(err) throw err;
+
+  });
+
+  socket.on('close', function(close){
+    if(close) server.close();
+    console.log('Connection Closed');
+  });
+});
+
+ee.on('@dm', function(client, string){
+  let nickname = client.nickname;
```

Comment:
You'll want nickname to be the first element in the array at the 0th index, so think about how you can get a string into the array and grab the first element from that array at index position 0. Here you aren't targeting the nickname correctly which is why the @dm is displayed to the specified client.